### PR TITLE
Refinement Improvements

### PR DESCRIPTION
**Pull request by shalvah-gs**
- Support passing `with` to the refinement. So we can do:
  ```ruby
  insurance_change_requests.crystallize(with: :insurance_collection)
  ```
  instead of the less clear:

 
  ```ruby
  insurance_change_requests.map(&:insurance_collection).crystallize
  ```

- Call ambiguity handler with resolved collection rather than original collection. This way we can avoid potentially logging huge values, while surfacing the most important information to the user (the actual values that differed).

- Add `crystallize` alias because the misspelling bugs me 😅

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -317,6 +316,24 @@ using NxtSupport::Refinements::Crystalizer
...
+NxtSupport::Crystalizer.new(collection: insurances, with: :effective_at).call
+NxtSupport::Crystalizer.new(collection: insurances, with: ->(i){ i.effective_at }).call
+insurances.crystalize(with: :effective_at)
+```
+
+The crystallizer raises a `NxtSupport::Crystalizer::Error` if the values are not the same, but you can override this:
+
+```ruby
+insurances.crystalize(with: :effective_at) do |effective_ats|
+  raise SomethingsWrong, "Insurances don't start at the same date"
```
Comment (AkihikoITOH) [inline README.md]:
```suggestion
  raise SomethingsWrong, "Insurances don't start on the same date"
```
(or, "at the same time")

**Review by AkihikoITOH — COMMENTED** [review]
"crystalize" isn't a misspelling but just a variant among "crystallize", "crystalise", and "crystallise".
I'm inclined to not add the alias because it could just be confusing when the class is named "Crystalizer". Also, having both in a codebase would make grepping harder. Not strongly against it though.

**Comment by shalvah-gs** [comment]
Interesting. I've never seen the single-l variant, but that's fair. Removed the alias, updated.

**Review by xgotyou — APPROVED** [review]
Like upgraded syntax! 👍

**Pull request merged**